File: app/src/main/java/com/kakacat/minitool/cleanfile/model/ScanTask.java
package com.kakacat.minitool.cleanfile.model;

import java.io.File;
import java.util.List;
import java.util.concurrent.Callable;

public class ScanTask implements Callable<Void> {

    private List<File> targetScanFileList;
    private List<List<FileItem>> fileListList;

    public ScanTask(List<File> targetScanFileList, List<List<FileItem>> fileListList) {
        this.targetScanFileList = targetScanFileList;
        this.fileListList = fileListList;
    }

    @Override
    public Void call() {
        targetScanFileList.forEach(this::start);
        return null;
    }

    private void start(File file) {
        if (file.isFile()) {
            if (file.length() == 0) {
                fileListList.get(1).add(new FileItem(file, false));
            } else {
                String fileName = file.getName();
                if (file.length() > 30 * 1024 * 1024) {
                    fileListList.get(0).add(new FileItem(file, false));
                }
                if (fileName.endsWith(".apk")) {
                    fileListList.get(2).add(new FileItem(file, false));
                } else if (isVideo(fileName)) {
                    fileListList.get(3).add(new FileItem(file, false));
                } else if (isAudio(fileName)) {
                    fileListList.get(4).add(new FileItem(file, false));
                }
            }
        } else if (file.isDirectory()) {
            File[] files = file.listFiles();
            if (files == null || files.length == 0) {
                fileListList.get(1).add(new FileItem(file, false));
            } else {
                for (File file1 : files) {
                    start(file1);
                }
            }
        }
    }

    private boolean isVideo(String fileName) {
        String[] videoSuffixes = {".mp4", ".mkv", ".flv"};

        for (String suffix : videoSuffixes) {
            if (fileName.endsWith(suffix)) {
                return true;
            }
        }

        return false;
    }

    private boolean isAudio(String fileName) {
        String[] audioSuffixes = {".mp3", ".flac", ".aac"};

        for (String suffix : audioSuffixes) {
            if (fileName.endsWith(suffix)) {
                return true;
            }
        }

        return false;
    }


}

File: app/src/main/java/com/kakacat/minitool/cleanfile/model/ScanTask.kt
package com.kakacat.minitool.cleanfile.model

import java.io.File
import java.util.function.Consumer

class ScanTask(private val targetScanFileList: MutableList<File>, private val fileListList: MutableList<MutableList<FileItem>>){

    fun start() {
        targetScanFileList.forEach(Consumer { file: File -> scan(file) })
    }

    private fun scan(file: File) {
        if (file.isFile) {
            if (file.length() == 0L) {
                fileListList[1].add(FileItem(file, false))
            } else {
                if (file.length() > 30 * 1024 * 1024) {
                    fileListList[0].add(FileItem(file, false))
                }
                val fileName = file.name
                when {
                    fileName.endsWith(".apk") -> {
                        fileListList[2].add(FileItem(file, false))
                    }
                    isVideo(fileName) -> {
                        fileListList[3].add(FileItem(file, false))
                    }
                    isAudio(fileName) -> {
                        fileListList[4].add(FileItem(file, false))
                    }
                }
            }
        } else if (file.isDirectory) {
            val files = file.listFiles()
            if (files == null || files.isEmpty()) {
                fileListList[1].add(FileItem(file, false))
            } else {
                for (file1 in files) {
                    scan(file1)
                }
            }
        }
    }

    private fun isVideo(fileName: String): Boolean {
        for (suffix in videoSuffixes) {
            if (fileName.endsWith(suffix)) {
                return true
            }
        }
        return false
    }

    private fun isAudio(fileName: String): Boolean {
        for (suffix in audioSuffixes) {
            if (fileName.endsWith(suffix)) {
                return true
            }
        }
        return false
    }

    companion object {
        private val videoSuffixes = arrayOf(".mp4", ".mkv", ".flv")
        private val audioSuffixes = arrayOf(".mp3", ".flac", ".aac")
    }
}